### PR TITLE
feat(gulp): add Gulp support

### PR DESCRIPTION
**Pull request by Awk34**
Needs:

- [ ] Grunt parity
 - [x] serve (needs testing)
 - [x] build (needs testing)
 - [x] watch (needs testing)
 - [ ] test
- [ ] tests

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -83,6 +83,18 @@ var AngularFullstackGenerator = yeoman.generators.Base.extend({
...
+          choices: [ 'Grunt', 'Gulp', 'Both'],
+          filter: function(val) {
+            return {
+              'Grunt': 'grunt',
+              'Gulp': 'gulp',
+              'Both': 'grunt_and_gulp'
+            }[val];
+          }
+        }, {
+          type: "list",
```
Comment (kingcody) [inline app/index.js]: What do you think about moving this to [`projectPrompts`](https://github.com/DaftMonk/generator-angular-fullstack/blob/canary/app/index.js#L244-L296)?

Comment (Awk34) [inline app/index.js]: that would probably make more sense XD

File: app/index.js [hunk 2]
```diff
@@ -82,7 +82,7 @@ var AngularFullstackGenerator = yeoman.generators.Base.extend({
       this.log('# Client\n');
 
       this.prompt([{
-          type: 'list',
+          type: "list",
```
Comment (kingcody) [inline app/index.js]: Should be single quotes here.

**Comment by david-mohr** [comment]
I was testing out your new gulp file and here are the changes I made to get `gulp build` working (yo 1.5.0):

```
--- gulpfile.babel.js.orig	2015-12-08 22:21:04.000000000 +1100
+++ gulpfile.babel.js	2015-12-08 22:15:37.000000000 +1100
@@ -4,6 +4,7 @@
 import _ from 'lodash';
 import del from 'del';
 import gulp from 'gulp';
+import babel from 'gulp-babel';
 import path from 'path';
 import gulpLoadPlugins from 'gulp-load-plugins';
 import http from 'http';
@@ -110,7 +111,11 @@

 let transpile = lazypipe()
     .pipe(plugins.sourcemaps.init)
-    .pipe(plugins.babel)
+    .pipe(babel, {
+      optional: [
+        'es7.classProperties'
+      ]
+    })
     .pipe(plugins.sourcemaps.write, '.');

 /********************
@@ -379,7 +384,7 @@

     let assets = plugins.useref.assets({searchPath: ['client', '.tmp']});

-    return gulp.src(paths.mainView)
+    return gulp.src(paths.client.mainView)
         .pipe(assets)
             .pipe(appFilter)
                 .pipe(plugins.addSrc.append('.tmp/templates.js'))
```

And in case you need them, here are my build selections.
```
html
sass
ngroute
bootstrap: true
UI bootstrap: true
mongoose: true
auth boilerplate: true
oAuth: none
socket.io: true
Jasmine
```
Thanks for all the great work!

**Comment by Awk34** [comment]
@david-mohr if you think there's a problem, feel free to submit a PR

**Comment by tmundt** [comment]
So, GULP is implemented?

**Comment by david-mohr** [comment]
I'm currently working on another PR based on my testing and porting some more Grunt tasks.

**Comment by Awk34** [comment]
@david-mohr awesome, I look forward to it

**Comment by david-mohr** [comment]
#1501

**Pull request merged**